frontend/src/context/WalletContext.tsx

```diff
@@ -0,0 +1,48 @@
+import { BigNumber } from "ethers"
+import { createContext, useContext, useEffect, useState } from "react"
+import { getTokenBalance } from "../sdk"
+import { GeyserContext } from "./GeyserContext"
+import Web3Context from "./Web3Context"
+
+export const WalletContext = createContext<{
+  walletAmount: BigNumber,
+  refreshWalletAmount: () => void,
+}>({
+  walletAmount: BigNumber.from('0'),
+  refreshWalletAmount: () => {},
+})
+
+export const WalletContextProvider: React.FC = ({ children }) => {
+  const [walletAmount, setWalletAmount] = useState<BigNumber>(BigNumber.from('0'))
+  const { signer } = useContext(Web3Context)
+  const { selectedGeyser } = useContext(GeyserContext)
+
+  const getWalletAmount = async () => {
+    if (selectedGeyser && signer) {
+      const { stakingToken } = selectedGeyser
+      try {
+        return getTokenBalance(stakingToken, await signer.getAddress(), signer)
+      } catch (e) {
+        console.error(e)
+      }
+    }
+  }
+
+  const refreshWalletAmount = () => getWalletAmount()
+
+  useEffect(() => {
+    let mounted = true
+    getWalletAmount().then(value => {
```

Comment:
we can wrap this in a try/catch and use async await instead

Reply:
q: why do we need to use an explicit `mounted` variable here

Reply:
Uses try/catch async/await now. The `mounted` variable is to fix the warning saying that a [state update cannot be performed on unmounted component](https://stackoverflow.com/questions/53949393/cant-perform-a-react-state-update-on-an-unmounted-component). Basically, being more explicit that the state update should only be dispatched if the component is mounted

Reply:
ooh interesting... ideally we'd be able to pass something like abortcontroller (for fetch) to the request, but this is the current workaround. 

sweet! would be great to have the link in a comment there mentioning that this is a workaround

Reply:
added comment and link